src/caja-desktop-window.c

```diff
@@ -252,34 +331,44 @@ unrealize (GtkWidget *widget)
 static void
 set_wmspec_desktop_hint (GdkWindow *window)
 {
-    GdkAtom atom;
+    GdkDisplay *display;
+    display = gdk_screen_get_display (gdk_screen_get_default());
+    if (GDK_IS_X11_DISPLAY (display))
```

Comment:
maybe
```suggestion
    if (GDK_IS_X11_WINDOW (window))
```

Reply:
Actually the whole function only does X11 stuff, so maybe only use it there?  or return early instead of adding an additional indentation layer?

Reply:
Using  `if (GDK_IS_X11_WINDOW (window)) ` let the build finish without a warning, but caused one of those functions not to run in x11, which allowed the desktop to be rendered over other windows, no longer always on the bottom. Probably not getting the actual window

Reply:
> Using `if (GDK_IS_X11_WINDOW (window)) ` let the build finish without a warning, but caused one of those functions not to run in x11, which allowed the desktop to be rendered over other windows, no longer always on the bottom. Probably not getting the actual window.

Okay, so that wasn't a so great idea in the end.  Weird, but I trust the test.